UIforETW/UIforETWDlg.cpp

```diff
@@ -40,10 +41,821 @@ const int kRecordTraceHotKey = 1234;
+
+const wchar_t btCswitchStacksString[ ] = 
+	L"This enables recording of call stacks on context switches, from both "
+	L"the thread being switched in and the readying thread. "
+	L"This should only be disabled if the performance "
+	L"of functions like WaitForSingleObject and "
+	L"SetEvent appears to be distorted, which can happen when the "
+	L"context-switch rate is very high.";
+
+const wchar_t InputTipString[ ] =
+	L"Input tracing inserts custom ETW events into "
+	L"traces which can be helpful when "
+	L"investigating performance problems that are correlated with user input. "
+	L"The default setting of "
+	L"'private' records alphabetic keys as 'A' and numeric keys as '0'. "
+	L"The 'full' setting records alphanumeric details. "
+	L"Both 'private' and 'full' record mouse movement and button clicks. The "
+	L"'off' setting records no input.";
+
+const wchar_t SampledStacksString[ ] = 
+	L"This enables recording of call stacks on CPU sampling events, which "
+	L"by default happen at 1 KHz. This should rarely be disabled.";
+
+const wchar_t SampleRateString[ ] = 
+	L"Checking this changes the CPU sampling frequency from the default of "
+	L"~1 KHz to the maximum speed of ~8 KHz. "
+	L"This increases the data rate and thus the size of traces "
+	L"but can make investigating brief CPU-bound "
+	L"performance problems (such as a single long frame) "
+	L"more practical.";
+
+const wchar_t GPUTracingString[ ] =
+	L"Check this to allow seeing GPU usage "
+	L"in WPA, and more data in GPUView.";
+
+const wchar_t ShowCommandsString[ ] =
+	L"This tells UIforETW to display the commands being "
+	L"executed. This can be helpful for diagnostic purposes "
+	L"but is not normally needed.";
+
+const wchar_t TracingModeString[ ] =
+	L"Select whether to trace straight to disk or to in-memory circular buffers.";
+
+const wchar_t TracesString[ ] =
+	L"This is a list of all traces found in %etwtracedir%, which defaults to "
+	L"documents\\etwtraces.";
+
+const wchar_t TraceNotesString[ ] =
+	L"Trace notes are intended for recording information about ETW traces, such "
+	L"as an analysis of what was discovered in the trace. "
+	L"Trace notes are auto-saved to a parallel text "
+	L"file - just type your analysis. "
+	L"The notes files will be renamed when you rename traces "
+	L"through the trace-list context menu.";
+
+const char NtSymbolEnvironmentVariableName[ ] =
+	"_NT_SYMBOL_PATH";
+
+const char NtSymCacheEnvironmentVariableName[ ] =
+	"_NT_SYMCACHE_PATH";
+
+const char DefaultSymbolPath[ ] =
+	"SRV*c:\\symbols*http://msdl.microsoft.com/download/symbols";
+
+const char ChromiumSymbolPath[ ] =
+	"SRV*c:\\symbols*http://msdl.microsoft.com/download/symbols;"
+	"SRV*c:\\symbols*https://chromium-browser-symsrv.commondatastorage.googleapis.com";
+
+const char DefaultSymbolCachePath[ ] =
+	"c:\\symcache";
+
+void handleUnexpectedCreateDirectory( _In_ const std::wstring& path, _In_ const DWORD lastErr )

```

Comment:
These utility functions should go in utility.cpp to avoid cluttering the main source file - if they are needed at all.